lib/fast-path.ts

```diff
@@ -324,9 +324,6 @@ FPp.needsParens = function (assumeExpressionContext) {
   }
 
   const parent = this.getParentNode();
-  if (!parent) {
-    return false;
-  }
```

Comment:
We delay this check so that we can reprint `(function () {} ())`, which otherwise will always have its outermost parens stripped since it has no parent.